ros2_batch_job/__main__.py

```diff
@@ -333,6 +335,30 @@ def process_coverage(args, job):
     return 0
 
 
+def force_xunit2_in_pytest_ini_files():
+    for path in Path('.').rglob('pytest.ini'):
+        config = configparser.ConfigParser()
+        config.read(str(path))
+        if StrictVersion(pytest.__version__) < StrictVersion('6.0.0'):
+            try:
+                # only if xunit2 is set continue the loop with the file unpatched
+                if config.get('pytest', 'junit_family') == 'xunit2':
+                    continue
+            except configparser.NoOptionError:
+                pass
+        else:
+            try:
+                # only need to correct explicit legacy existing option
+                if config.get('pytest', 'junit_family') != 'legacy':
+                    continue
+            except configparser.NoOptionError:
+                continue
```

Comment:
I am wondering if the duplicate logic in the `if` and `else` block can't be avoided...

Reply:
the code was awful, yes. I made some changes in 057c1f7 . Hopefully it can be read more easily.